react/features/settings/components/native/ProfileView.tsx

```diff
@@ -48,6 +64,50 @@ const ProfileView = ({ isInWelcomePage }: { isInWelcomePage?: boolean; }) => {
     },
     [ dispatch, updateSettings, email, displayName ]);
 
+    const onLogin = useCallback(() => {
+        dispatch(login());
+    }, [ logoutUrl ]);
+
+    const onLogout = useCallback(() => {
+        dispatch(logout());
+    }, [ logoutUrl ])
+
+    const headerLeft = () => (
+        <HeaderNavigationButton
+            color = { BaseThemeNative.palette.link01 }
+            onPress = { goBack }
+            src = { IconArrowLeft }
+            style = { styles.backBtn }
+            twoActions = { true } />
+    );
+
+    const headerRight = () => {
+        if (logoutUrl) {
```

Comment:
The URL will exist even before you login so looks like this logic won't work? You need to know if you are logged in or not. Also, the URL works for the token case, but the normal XMPP auth also supports logout.

Reply:
Updated. Used authLogin from store.